diff --git a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/YandexMobileAdsModule.kt b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/YandexMobileAdsModule.kt
index b64df07..5d869ad 100644
--- a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/YandexMobileAdsModule.kt
+++ b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/YandexMobileAdsModule.kt
@@ -20,7 +20,7 @@ class YandexMobileAdsModule(
 
     @ReactMethod
     fun showDebugPanel(promise: Promise) {
-        currentActivity?.let { activity ->
+        reactContext.currentActivity?.let { activity ->
             activity.runOnUiThread {
                 MobileAds.showDebugPanel(activity)
                 promise.resolve(null)
diff --git a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/app_open/YandexAppOpenAdModule.kt b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/app_open/YandexAppOpenAdModule.kt
index d517133..6631c87 100644
--- a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/app_open/YandexAppOpenAdModule.kt
+++ b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/app_open/YandexAppOpenAdModule.kt
@@ -41,7 +41,7 @@ internal class YandexAppOpenAdModule(
 
     @ReactMethod
     fun newLoader(promise: Promise) {
-        currentActivity?.runOnUiThread {
+        context.currentActivity?.runOnUiThread {
             val loader = AppOpenAdLoader(reactApplicationContext)
             val loaderId = loadersStorage.put(loader)
             loader.setAdLoadListener(RNAppOpenAdLoadListener(loaderId))
@@ -55,7 +55,7 @@ internal class YandexAppOpenAdModule(
         adRequestConfiguration: ReadableMap,
         promise: Promise,
     ) {
-        currentActivity?.runOnUiThread {
+        context.currentActivity?.runOnUiThread {
             val loader = loadersStorage.get(loaderId)
             if (loader == null) {
                 promise.reject(PluginError.ObjectNotFound)
@@ -84,14 +84,14 @@ internal class YandexAppOpenAdModule(
         adObjectId: String,
         promise: Promise,
     ) {
-        val activity = currentActivity
+        val activity = context.currentActivity
         if (activity == null) {
             deleteAd(adObjectId)
             promise.reject(PluginError.CurrentActivityIsNull)
             return
         }
         activity.runOnUiThread {
-            if (currentActivity == null || currentActivity?.isFinishing == true) {
+            if (context.currentActivity == null || context.currentActivity?.isFinishing == true) {
                 deleteAd(adObjectId)
                 promise.reject(PluginError.CurrentActivityIsNull)
                 return@runOnUiThread
diff --git a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/interstitial/YandexInterstitialAdModule.kt b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/interstitial/YandexInterstitialAdModule.kt
index 9ebf5d0..31b6e8d 100644
--- a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/interstitial/YandexInterstitialAdModule.kt
+++ b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/interstitial/YandexInterstitialAdModule.kt
@@ -41,7 +41,7 @@ internal class YandexInterstitialAdModule(
 
     @ReactMethod
     fun newLoader(promise: Promise) {
-        currentActivity?.runOnUiThread {
+        context.currentActivity?.runOnUiThread {
             val loader = InterstitialAdLoader(reactApplicationContext)
             val loaderId = loadersStorage.put(loader)
             loader.setAdLoadListener(RNInterstitialAdLoadListener(loaderId))
@@ -55,7 +55,7 @@ internal class YandexInterstitialAdModule(
         adRequestConfiguration: ReadableMap,
         promise: Promise,
     ) {
-        currentActivity?.runOnUiThread {
+        context.currentActivity?.runOnUiThread {
             val loader = loadersStorage.get(loaderId)
             if (loader == null) {
                 promise.reject(PluginError.ObjectNotFound)
@@ -84,14 +84,14 @@ internal class YandexInterstitialAdModule(
         adObjectId: String,
         promise: Promise,
     ) {
-        val activity = currentActivity
+        val activity = context.currentActivity
         if (activity == null) {
             deleteAd(adObjectId)
             promise.reject(PluginError.CurrentActivityIsNull)
             return
         }
         activity.runOnUiThread {
-            if (currentActivity == null || currentActivity?.isFinishing == true) {
+            if (context.currentActivity == null || context.currentActivity?.isFinishing == true) {
                 deleteAd(adObjectId)
                 promise.reject(PluginError.CurrentActivityIsNull)
                 return@runOnUiThread
diff --git a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/rewarded/YandexRewardedAdModule.kt b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/rewarded/YandexRewardedAdModule.kt
index 678ad2d..6f693ba 100644
--- a/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/rewarded/YandexRewardedAdModule.kt
+++ b/node_modules/yandex-mobile-ads/android/src/main/java/com/yandexmobileads/rewarded/YandexRewardedAdModule.kt
@@ -43,7 +43,7 @@ internal class YandexRewardedAdModule(
 
     @ReactMethod
     fun newLoader(promise: Promise) {
-        currentActivity?.runOnUiThread {
+        context.currentActivity?.runOnUiThread {
             val loader = RewardedAdLoader(reactApplicationContext)
             val loaderId = loadersStorage.put(loader)
             loader.setAdLoadListener(RNRewardedAdLoadListener(loaderId))
@@ -57,7 +57,7 @@ internal class YandexRewardedAdModule(
         adRequestConfiguration: ReadableMap,
         promise: Promise,
     ) {
-        currentActivity?.runOnUiThread {
+        context.currentActivity?.runOnUiThread {
             val loader = loadersStorage.get(loaderId)
             if (loader == null) {
                 promise.reject(PluginError.ObjectNotFound)
@@ -86,14 +86,14 @@ internal class YandexRewardedAdModule(
         adObjectId: String,
         promise: Promise,
     ) {
-        val activity = currentActivity
+        val activity = context.currentActivity
         if (activity == null) {
             deleteAd(adObjectId)
             promise.reject(PluginError.CurrentActivityIsNull)
             return
         }
         activity.runOnUiThread {
-            if (currentActivity == null || currentActivity?.isFinishing == true) {
+            if (context.currentActivity == null || context.currentActivity?.isFinishing == true) {
                 deleteAd(adObjectId)
                 promise.reject(PluginError.CurrentActivityIsNull)
                 return@runOnUiThread
